cardinal/world.go

```diff
@@ -49,7 +49,7 @@ const (
 type World struct {
 	server          *server.Server
 	tickChannel     <-chan time.Time
-	tickDoneChannel chan<- uint64
+	tickDoneChannel chan types.TickResult
```

Comment:
nit: `chan<- types.TickResult` specifies a channel that can only have data pushed to it, which is how World should be using the tickDone channel.

Reply:
I'm reading TODO comment from @smsunarto at line `219` before changes : 
> One thing we need to do is to make the tickDone channel return the tick number AND error if any so it can be used to check if there are errors when trying to tick.

But if there is no one using this error in the future i'll change it to send only channel `chan<- types.TickResultinstead` instead of bidirectional channel `chan types.TickResult`

Reply:
reverted to `chan<- uint64` because tick always throw panic if there is any error